fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/RangerHiveAccessController.java

```diff
@@ -0,0 +1,188 @@
+
+    public RangerHiveAccessController(Map<String, String> properties) {
+        String serviceName = properties.get("ranger.service.name");
+        hivePlugin = new RangerHivePlugin(serviceName);
+    }
+
+    private RangerAccessRequestImpl createRequest(UserIdentity currentUser, HiveAccessType accessType) {
+        RangerAccessRequestImpl request = new RangerAccessRequestImpl();
+        request.setUser(currentUser.getQualifiedUser());
+        request.setUserRoles(currentUser.getRoles());
+        request.setAction(accessType.name());
+        if (accessType == HiveAccessType.USE) {
+            request.setAccessType(RangerPolicyEngine.ANY_ACCESS);
+        } else {
+            request.setAccessType(accessType.name().toLowerCase());
+        }
+        request.setClientIPAddress(currentUser.getHost());
+        request.setClientType(CLIENT_TYPE_DORIS);
+        request.setAccessTime(new Date());
+
+        return request;
+    }
+
+    public void checkPrivileges(UserIdentity currentUser, HiveAccessType accessType,
+                                List<RangerHiveResource> hiveResources) throws AuthorizationException {
+        List<RangerAccessRequest> requests = new ArrayList<>();
+        for (RangerHiveResource resource : hiveResources) {
+            RangerAccessRequestImpl request = createRequest(currentUser, accessType);
+            request.setResource(resource);
+
+            requests.add(request);
+        }
+
+        RangerHiveAuditHandler auditHandler = new RangerHiveAuditHandler(hivePlugin.getConfig());
+        Collection<RangerAccessResult> results = hivePlugin.isAccessAllowed(requests, auditHandler);
+        for (RangerAccessResult result : results) {
+            LOG.debug("match policy:" + result.getPolicyId());
+            LOG.debug("will audit: " + result.toString());
+            auditHandler.processResult(result);
+            if (!result.getIsAllowed()) {
+                LOG.warn(result.getReason());
+                throw new AuthorizationException(String.format(
+                    "Permission denied: user [%s] does not have privilege for [%s] command on [%s]",
+                    currentUser.getQualifiedUser(), accessType.name(),
+                    result.getAccessRequest().getResource().getAsString()));
+            }
+        }
+    }
+
+    public boolean checkPrivilege(UserIdentity currentUser, HiveAccessType accessType,
+                                  RangerHiveResource resource) {
+        RangerHiveAuditHandler auditHandler = new RangerHiveAuditHandler(hivePlugin.getConfig());
+        RangerAccessRequestImpl request = createRequest(currentUser, accessType);
+        request.setResource(resource);
+
+        RangerAccessResult result = hivePlugin.isAccessAllowed(request, auditHandler);
+        auditHandler.processResult(result);
+        auditHandler.flushAudit();
+
+        if (result == null) {
+            LOG.warn("Error getting authorizer result, please check your ranger config");
```

Comment:
Could we print more detail error info here?

Reply:
fixed